base_ebill_payment_contract/models/res_partner.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo import models
+
+_logger = logging.getLogger(__name__)
+
+
+class ResPartner(models.Model):
+
+    _inherit = "res.partner"
+
+    def get_active_contract(self, transmit_method):
+        """Return the active contract for a specific transmit method."""
+        self.ensure_one()
+        # FIXME: Search on non stored field
+        contracts = self.env["ebill.payment.contract"].search(
+            [("is_valid", "=", True), ("partner_id", "=", self.id)], limit=1,
```

Comment:
We have `Non-stored field ebill.payment.contract.is_valid cannot be searched.` in logs.

2 options:

* Implement a "search" method on the computed field
* Search all the contracts for the partner (remove limit) and filter them afterwards, it seems acceptable as the number of contract per partner will be a few. 

Solution 2 may be easier, but for UX, being able to search on the `is_valid` field may be useful if we have a list view with all the contracts.

Reply:
I implemented the search method on the third commit.